lib/Scheduler/aco.cpp

```diff
@@ -251,14 +253,59 @@ FUNC_RESULT ACOScheduler::FindSchedule(InstSchedule *schedule_out,
                                        SchedRegion *region) {
   rgn_ = region;
 
+  if(!aco_use_two_pass)
+  {
+    pass = ACOPass::ALL_IN_ONE;
+    return performSchedPass(schedule_out);
+  }
+  else
+  {
+    //in the first pass we schedule for register pressure only
+    //in the second pass we schedule for lower schedule length
+    //but only accept schedules with an RP <= the first passes RP
+    //the first pass result is the initial schedule the second pass
+    pass = ACOPass::FIRST;
+    FUNC_RESULT pass1Res = performSchedPass(schedule_out);
+	if(pass1Res!=RES_SUCCESS) {
+      return pass1Res;
+    }
+    pass = ACOPass::SECOND;
+    setInitialSched(schedule_out);
+    return performSchedPass(schedule_out);
+
+  }
+}
+
+//compares to schedules taking into account which pass
+//we are on and returns whether the old schedule
+//is inferior to the new one and ought to be replaced
+bool ACOScheduler::shouldReplaceSchedule(ACOPass pass, InstSchedule* oldSched,
+                                         InstSchedule* newSched) {
+  //Logger::Info("old_sched_addr:%p, new_sched_addr:%p", oldSched, newSched);
+  return newSched!=NULL &&
```

Comment:
This needs to be broken up into steps. Maybe with some static functions.

Reply:
I will make it static and look into breaking it up into steps/functions.  I didn't before, because I didn't want to add extra branches to fairly hot code

Reply:
> I will make it static and look into breaking it up into steps/functions. I didn't before, because I didn't want to add extra branches to fairly hot code

Let compiler do the optimization it will inline it. LLVM programming guide recommends this as well.